### PR TITLE
modify echo10-construction/src/main.py to include three additional at…

### PR DESCRIPTION
**Pull request by cirrusasf**
TODO:

- [x] Address `TODO` and `FIXME` in code
- [x] Should any more unit tests have `v3` counterparts?

### CODE REVIEW

File: echo10-construction/src/main.py [hunk 1]
```diff
@@ -119,13 +119,24 @@ def get_granule_data(inputs, config):
...
+            'VERSION': granule_metadata['version'],
         },
         'input_granules': input_granules,
         'visible': 'true',
         'orderable': 'true',
         'online_access_url': online_access_url,
         'browse_url': browse_url,
     }
+
+    if granule_metadata['temporal_baseline_days'] != -1:
```
Comment (asjohnston-asf) [inline echo10-construction/src/main.py]: we're not expecting sentinel values for these; they'll be optional in `sds_metadata`, so we just need to look for the existence of the key:
```suggestion
    if 'temporal_baseline_days' in granule_metadata:
```

File: echo10-construction/src/main.py [hunk 2]
```diff
@@ -119,13 +119,24 @@ def get_granule_data(inputs, config):
             'THUMBNAIL_URL': browse_url,
             'PERPENDICULAR_BASELINE': granule_metadata['perpendicular_baseline'],
             'MISSION_NAME': mission,
+            'VERSION': granule_metadata['version'],
```
Comment (asjohnston-asf) [inline echo10-construction/src/main.py]: The failing test points out that the version field is `sds_metadata['version']`, not `sds_metadata['metadata']['version']`
```suggestion
            'VERSION': sds_metadata['version'],
```

File: echo10-construction/src/main.py [hunk 3]
```diff
@@ -91,7 +91,7 @@ def get_granule_data(inputs, config):
         'beginning_date_time': granule_metadata['sensing_start'],
         'ending_date_time': granule_metadata['sensing_stop'],
         'orbits': granule_metadata['orbit_number'],
-        'platforms': [p.upper() for p in set(granule_metadata['platform'])],
+        'platforms': [p.upper() for p in granule_metadata['platform']],
```
Comment (jtherrmann) [inline echo10-construction/src/main.py]: Let's double-check that we don't need to worry about duplicate values by getting rid of `set`.

File: echo10-construction/src/echo10_construction.py [hunk 4]
```diff
@@ -91,8 +91,7 @@ def get_granule_data(inputs, config):
         'beginning_date_time': granule_metadata['sensing_start'],
         'ending_date_time': granule_metadata['sensing_stop'],
         'orbits': granule_metadata['orbit_number'],
-        # FIXME we may need to preserve order of platforms, so tests don't intermittently fail
-        'platforms': [p.upper() for p in set(granule_metadata['platform'])],
+        'platforms': sorted(set([p.upper() for p in granule_metadata['platform']])),
```
Comment (jtherrmann) [inline echo10-construction/src/echo10_construction.py]:
```suggestion
        'platforms': sorted(set(p.upper() for p in granule_metadata['platform'])),
```

**Comment by jtherrmann** [comment]
This just needs a Changelog entry, otherwise looks good to me. But I'd like @asjohnston-asf 's approval before merging.

File: echo10-construction/src/echo10_construction.py [hunk 5]
```diff
@@ -140,16 +151,21 @@ def create_granule_echo10_in_s3(inputs, config):
...
+
     upload_content_to_s3(echo10_s3_object, echo10_content)
 
-    del granule_data['size_mb_data_granule']
-    del granule_data['additional_attributes']['BYTES']
+    if 'size_mb_data_granule' in granule_data:
+        del granule_data['size_mb_data_granule']
+
+    if 'BYTES' in granule_data['additional_attributes']:
+        del granule_data['additional_attributes']['BYTES']
```
Comment (asjohnston-asf) [inline echo10-construction/src/echo10_construction.py:168]: I'm fine with being defensive here, but did anything in particular prompt the change? Is there a particular test is caused to fail?

**Review by asjohnston-asf — APPROVED** [review]

**Comment by asjohnston-asf** [comment]
I opened https://github.com/asfadmin/grfn-ingest/pull/202 to update the changelog to reflect the entirety of the next release, so I'm comfortable labeling this pr as `bumpless`.

**Pull request merged**